File: src/PlanGraph.js
class PlanGraph{
    constructor(){
        //keep track of adjacent nodes
        this.adjList = new Map();
        //keep track of relationships between nodes
        this.edgeRelations = new Map();
    }

    //add node to graph
    addNode(n){
        this.adjList.set(n,[]);
    }

    //plan graph is directed
    addEdge(n, nn, relation){
        this.AdjList.get(n).push(nn);
        this.edgeRelations.set(n, [nn, relation]);
    }
}

File: src/PlanGraph.cs
namespace graph{
    public class PlanGraph{
        List<Node> nodes { get; }

        public PlanGraph(){
            this.nodes = new List<Node>();
        }

        //add new node to graph
        public Node addNode(string codeblock, string instruction, string status){
            var newNode = new Node(new Tuple<string, string, string>(codeblock, instruction, status));
            nodes.Add(newNode);
            return newNode;
        }

        //add neighbor to node
        public void addNeighbor(Node x, Node y, string relationship){
            x.getNeighbors().Add(new Tuple<Node,string>(y, relationship));
        }
    }
}